File: app/src/main/java/com/juniperphoton/myerlistandroid/activity/StartActivity.java
package com.juniperphoton.myerlistandroid.activity;

import android.content.Intent;
import android.os.Bundle;
import android.support.annotation.Nullable;

import com.juniperphoton.myerlistandroid.R;
import com.juniperphoton.myerlistandroid.util.Params;

import butterknife.ButterKnife;
import butterknife.OnClick;

@SuppressWarnings("UnusedDeclaration")
public class StartActivity extends BaseActivity {
    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_start);
        ButterKnife.bind(this);
    }

    @OnClick(R.id.login_btn)
    void onClickLogin() {
        Intent intent = new Intent(this, LoginActivity.class);
        intent.putExtra(Params.LOGIN_MODE, Params.LoginMode.LOGIN);
        startActivity(intent);
    }

    @OnClick(R.id.register_btn)
    void onClickRegister() {
        Intent intent = new Intent(this, LoginActivity.class);
        intent.putExtra(Params.LOGIN_MODE, Params.LoginMode.REGISTER);
        startActivity(intent);
    }

    @OnClick(R.id.offline_btn)
    void onClickOffline() {

    }
}

File: app/src/main/java/com/juniperphoton/myerlist/activity/StartActivity.kt
package com.juniperphoton.myerlist.activity

import android.content.Intent
import android.os.Bundle
import butterknife.ButterKnife
import butterknife.OnClick
import com.juniperphoton.myerlist.R
import com.juniperphoton.myerlist.util.Params

@Suppress("unused","unused_parameter")
class StartActivity : BaseActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_start)
        ButterKnife.bind(this)
    }

    @OnClick(R.id.loginButton)
    fun onClickLogin() {
        val intent = Intent(this, LoginActivity::class.java)
        intent.putExtra(Params.LOGIN_MODE, Params.LoginMode.LOGIN)
        startActivity(intent)
    }

    @OnClick(R.id.register_btn)
    fun onClickRegister() {
        val intent = Intent(this, LoginActivity::class.java)
        intent.putExtra(Params.LOGIN_MODE, Params.LoginMode.REGISTER)
        startActivity(intent)
    }

    @OnClick(R.id.offline_btn)
    fun onClickOffline() {
    }
}
